### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by LysandreJik**
This PR removes deprecations that should be removed with v0.8.0.

### CODE REVIEW

File: src/huggingface_hub/README.md [hunk 1]
```diff
@@ -97,9 +97,9 @@ programmatic way of creating a repo, deleting it (`⚠️ caution`), pushing a
 single file to a repo or listing models from the Hub, you'll find helpers in
 `hf_api.py`. Some example functionality available with the `HfApi` class:
 
-* `login()`
+* set_access_token()
+* unset_access_token()
```
Comment (julien-c) [inline src/huggingface_hub/README.md:101]:
```suggestion
* `set_access_token()`
* `unset_access_token()`
```

**Review by julien-c — COMMENTED** [review]
looks good to me!

File: src/huggingface_hub/commands/user.py [hunk 2]
```diff
@@ -366,35 +335,9 @@ def login_token_event(t):
 
     token_finish_button.on_click(login_token_event)
 
-    def login_password_event(t):
```
Comment (osanseviero) [inline src/huggingface_hub/commands/user.py:369]: I love when code is deleted 🔥

File: tests/test_hf_api.py [hunk 3]
```diff
@@ -109,37 +106,36 @@ class HfApiCommonTest(unittest.TestCase):
 
 class HfApiLoginTest(HfApiCommonTest):
     def setUp(self) -> None:
-        erase_from_credential_store(USER)
+        erase_from_credential_store(USERNAME_PLACEHOLDER)
```
Comment (osanseviero) [inline tests/test_hf_api.py:109]: Why is this change needed?

**Review by osanseviero — APPROVED** [review]
Very nice PR 🔥 

From what I recall and quick check all downstream libraries should have pinned versions so it should not break anything, but let's keep an eye for things that could potentially break

File: tests/test_hf_api.py (hunk 3 above)
Comment (LysandreJik) [reply tests/test_hf_api.py:109]: Because beforehand we were saving the username/password in the credential cache, and that's not the case anymore!

Now we're saving `hf_user:token` instead, with `USERNAME_PLACEHOLDER` being `hf_user`.
